x/distribution/keeper/allocation.go

```diff
@@ -1,57 +1,85 @@
 package keeper
 
 import (
+	abci "github.com/tendermint/tendermint/abci/types"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	"github.com/cosmos/cosmos-sdk/x/distribution/types"
 )
 
-// Allocate fees handles distribution of the collected fees
-func (k Keeper) AllocateTokens(ctx sdk.Context, percentVotes sdk.Dec, proposer sdk.ConsAddress) {
-
-	// get the proposer of this block
-	proposerValidator := k.stakeKeeper.ValidatorByConsAddr(ctx, proposer)
+// allocate fees handles distribution of the collected fees
+func (k Keeper) AllocateTokens(ctx sdk.Context, sumPrecommitPower int64, totalPower int64, proposer sdk.ConsAddress, votes []abci.VoteInfo) {
 
-	proposerDist := k.GetValidatorDistInfo(ctx, proposerValidator.GetOperator())
+	// fetch collected fees & fee pool
+	feesCollectedInt := k.feeCollectionKeeper.GetCollectedFees(ctx)
+	feesCollected := sdk.NewDecCoins(feesCollectedInt)
+	feePool := k.GetFeePool(ctx)
 
-	// get the fees which have been getting collected through all the
-	// transactions in the block
-	feesCollected := k.feeCollectionKeeper.GetCollectedFees(ctx)
-	feesCollectedDec := types.NewDecCoins(feesCollected)
+	// clear collected fees, which will now be distributed
+	k.feeCollectionKeeper.ClearCollectedFees(ctx)
 
-	feePool := k.GetFeePool(ctx)
-	// Temporary workaround to keep CanWithdrawInvariant happy.
-	// General discussions here: https://github.com/cosmos/cosmos-sdk/issues/2906#issuecomment-441867634
-	if k.stakeKeeper.GetLastTotalPower(ctx).IsZero() {
-		feePool.CommunityPool = feePool.CommunityPool.Plus(feesCollectedDec)
+	// temporary workaround to keep CanWithdrawInvariant happy
+	// general discussions here: https://github.com/cosmos/cosmos-sdk/issues/2906#issuecomment-441867634
```

Comment:
Is this still applicable?

Reply:
We should never run into this case in practice, but the code path is safe I think.